src/Maestro/tests/Maestro.ScenarioTests/ObjectHelpers/SubscriptionBuilder.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.DotNet.Darc;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Newtonsoft.Json.Linq;
+
+namespace Maestro.ScenarioTests.ObjectHelpers
+{
+    public class SubscriptionBuilder
+    {
+        /// <summary>
+        /// Creates a subscription object based on a standard set of test inputs
+        /// </summary>
+        public static Subscription BuildSubscription(string repo1Uri, string repo2Uri, string targetBranch, string channelName, string subscriptionId,
+            UpdateFrequency updateFrequency, bool batchable, List<string> mergePolicyNames, List<string> ignoreChecks)
+        {
+            Subscription expectedSubscription = new Subscription(Guid.Parse(subscriptionId), true, repo1Uri, repo2Uri, targetBranch);
+            expectedSubscription.Channel = new Channel(42, channelName, "test");
+            expectedSubscription.Policy = new SubscriptionPolicy(batchable, updateFrequency);
+
+            List<MergePolicy> mergePolicies = new List<MergePolicy>();
+
+            if (mergePolicyNames != null)
+            {
+
+                if (mergePolicyNames.Contains(Constants.StandardMergePolicyName))
+                {
+                    mergePolicies.Add(new MergePolicy
+                    {
+                        Name = Constants.StandardMergePolicyName
+                    });
+                }
+
+                if (mergePolicyNames.Contains(Constants.NoExtraCommitsMergePolicyName))
+                {
+                    mergePolicies.Add(
+                        new MergePolicy
+                        {
+                            Name = Constants.NoExtraCommitsMergePolicyName
+                        });
+                }
+
+                if (mergePolicyNames.Contains(Constants.AllCheckSuccessfulMergePolicyName) && ignoreChecks.Any())
+                {
+                    mergePolicies.Add(
+                        new MergePolicy
+                        {
+                            Name = Constants.AllCheckSuccessfulMergePolicyName,
+                            Properties = ImmutableDictionary.Create<string, JToken>()
+                                .Add(Constants.IgnoreChecksMergePolicyPropertyName, JToken.FromObject(ignoreChecks))
+                        });
+                }
+
+                if (mergePolicyNames.Contains(Constants.NoRequestedChangesMergePolicyName))
```

Comment:
Are the "merge policys" a fixed, enumerable set?  If so... it should probably be an enumeration rather than a bunch of strings.  And if we can have more than once, then it can be a "[Flags]" enum rather than a List<MergePolicies>.  It seems like the are based on this code, and then you don't have to test for what happens if some passed "Pizza" into this method. :-D

Reply:
To clarify, are you suggesting adding the enum for the test code or in Darc? Darc uses the strings & constants.

Reply:
Well, really I'd like darc not to use a bunch of magic strings.  But that's probably a bigger change.  We should probably log a bug for it that can get tossed into the void. :-)